% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/explain_last_error.R
\name{explain_last_error}
\alias{explain_last_error}
\title{Explain Last Error Using OpenAI}
\usage{
explain_last_error(
  model = "gpt-4",
  prompt =
    "Give an explanation for the following error in the R programing language and use the provided stack trace to aid the user in resolving the problem. Do not ask for code snippets or any further information.",
  show_trace = F,
  show_trace_JSON = F
)
}
\arguments{
\item{model}{A string specifying the type of model to be used, default is "gpt-4".}

\item{prompt}{A string defining the prompt for the explanation. The default explains a provided error using a stack trace.}

\item{show_trace}{A boolean defining if the function should print the trace from the last error, default is False.}

\item{show_trace_JSON}{A boolean defining if the function should print the JSON representation of the trace from the last error, default is False.}
}
\value{
This function primarily returns its results to console via message(). It does not return anything to the global scope unless assigned to a variable. In that case it returns a string explaining the nature of the provided error. It includes an error message and an explanation generated by the specified model.
}
\description{
Explain Last Error Using OpenAI
}
\note{
This function relies on enriched errors provided by the rlang package. If such an error is not available it will attach rlang and prompt the user to rerun their command to generate an enriched error.
}
\examples{
\dontrun{
  library(rlang)
  global_entrace()
  # Run a command that produces an error
  explain_last_error(show_trace = T, show_trace_JSON = T)
}

}
\author{
Nathan C. Layman
}
